infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LRRollingUpgradeHandler.java

```diff
@@ -47,28 +58,37 @@ public LRRollingUpgradeHandler(ILogReplicationVersionAdapter versionAdapter) {
     }
 
     public boolean isLRUpgradeInProgress(TxnContext txnContext) {
-        if (isClusterAllAtV2) {
+        try {
+            // If LOG_REPLICATION_PLUGIN_VERSION_TABLE exists, it indicates an upgrade from a
+            // previous version was performed. It has since been removed in the current version.
+            txnContext.getTable(LOG_REPLICATION_PLUGIN_VERSION_TABLE);
+
```

Comment:
Shouldn't this check against RegistryTable directly? otherwise if the runtime associated with this txn hasn't opened this table in the in-memory `tableMap`, it still throws `IllegalArgumentException` and we are not catching it right?

Reply:
And also a question for it, I remember George was working on drop table support in Corfu, should we consider dropping this table after upgrade? (for setups upgrading from a version with this table).  or maybe something could be added to backlog?

Reply:
For the second question, @pankti-m, @hisundar, and I had discussed this also. The problem with dropping the table after upgrade is that that if there's ever a rollback to a prior version we will hit a trimmed exception since the drop table is a trim operation with no checkpoint to revert too, and the logic to recover from this will not be present in a prior version.

Reply:
+1

Reply:
@xcchang , valid point about IllegalArgumentException.  This is being addressed in the next PR - https://github.com/CorfuDB/CorfuDB/pull/3596.  The current tests for upgrade are not restarting the nodes so it is not being hit.  The next PR changes this workflow and handles the other exception.